src/widgets/inspector.rs

```diff
@@ -0,0 +1,88 @@
+use kayak_core::styles::{PositionType, Style, StyleProp, Units};
+use kayak_core::{Bound, Color, EventType, OnEvent, VecTracker};
+use kayak_render_macros::{constructor, use_state};
+
+use crate::core::derivative::*;
+use crate::core::{rsx, widget, MutableBound};
+
+use crate::widgets::{Background, Button, Text};
+
+#[derive(Clone, PartialEq)]
+pub enum InspectData {
+    None,
+    Data(Vec<String>),
+}
+
+#[widget]
+pub fn Inspector() {
+    let (inspect_data, set_inspect_data, _) = use_state!(Vec::<String>::new());
+
+    let background_styles = Some(Style {
+        background_color: StyleProp::Value(Color::new(0.125, 0.125, 0.125, 1.0)),
+        border_radius: StyleProp::Value((0.0, 0.0, 0.0, 0.0)),
+        position_type: StyleProp::Value(PositionType::SelfDirected),
+        left: StyleProp::Value(Units::Stretch(1.0)),
+        top: StyleProp::Value(Units::Stretch(0.0)),
+        bottom: StyleProp::Value(Units::Stretch(0.0)),
+        width: StyleProp::Value(Units::Pixels(200.0)),
+        ..styles.clone().unwrap_or_default()
+    });
+
+    let last_clicked = context.get_last_clicked_widget();
+    context.bind(&last_clicked);
+
+    let last_clicked_value = last_clicked.get();
+    let (id, _) = last_clicked_value.into_raw_parts();
+
+    let mut parent_id_move = None;
+    if let Some(layout) = context.widget_manager.get_layout(&last_clicked_value) {
+        if let Some(node) = context.widget_manager.get_node(&last_clicked_value) {
+            let mut data = Vec::new();
+
+            if let Some(name) = context.widget_manager.get_name(&last_clicked_value) {
+                data.push(format!("Name: {}", name));
+            }
+
+            data.push(format!("ID: {}", id));
+            data.push(format!("X: {}", layout.posx));
+            data.push(format!("Y: {}", layout.posy));
+            data.push(format!("Width: {}", layout.width));
+            data.push(format!("Height: {}", layout.height));
+            data.push(format!("RenderCommand: {:?}", node.styles.render_command));
+            data.push(format!("Height: {:?}", node.styles.height));
```

Comment:
Does `Text` support newlines? What happens if you use `"{:#?}"`?

Though, I guess that might take up too much space...

Reply:
I'm fine with using `{:#?}` if that works.

Reply:
Eventually I want to print out all of styles, but that can happen later..

Reply:
You could also maybe put this data in separate `Fold` widgets to help break things up (assuming they work properly haha)